presto-spi/src/main/java/com/facebook/presto/spi/block/BlockUtil.java

```diff
@@ -216,4 +228,88 @@ public static boolean internalPositionInRange(int internalPosition, int offset,
         assert withinRange : format("internalPosition %s is not within range [%s, %s)", internalPosition, offset, positionCount + offset);
         return withinRange;
     }
+
+    static boolean[] appendNullToIsNullArray(boolean[] isNull, int offsetBase, int positionCount)
+    {
+        checkArrayRange(isNull, offsetBase, positionCount);
+
+        int desiredLength = offsetBase + positionCount + 1;
+        boolean[] newIsNull = new boolean[desiredLength];
+        if (isNull != null) {
+            System.arraycopy(isNull, 0, newIsNull, 0, desiredLength - 1);
+        }
+        newIsNull[desiredLength - 1] = true;
+        return newIsNull;
+    }
+
+    static int[] appendNullToOffsetsArray(int[] offsets, int offsetBase, int positionCount)
+    {
+        checkArrayRange(offsets, offsetBase, positionCount + 1);
+
+        int desiredLength = offsetBase + positionCount + 2;
+        int[] newOffsets = Arrays.copyOf(offsets, desiredLength);
+        newOffsets[desiredLength - 1] = newOffsets[desiredLength - 2];
+        return newOffsets;
+    }
+
+    public static byte[] ensureCapacity(byte[] buffer, int capacity)
+    {
+        if (buffer == null) {
+            buffer = new byte[capacity];
+        }
+        else if (buffer.length < capacity) {
+            buffer = java.util.Arrays.copyOf(buffer, capacity);
+        }
+
+        return buffer;
+    }
+
+    public static short[] ensureCapacity(short[] buffer, int capacity)
+    {
+        if (buffer == null) {
+            buffer = new short[capacity];
+        }
+        else if (buffer.length < capacity) {
+            buffer = java.util.Arrays.copyOf(buffer, capacity);
+        }
+
+        return buffer;
+    }
+
+    public static int[] ensureCapacity(int[] buffer, int capacity)
+    {
+        if (buffer == null) {
+            buffer = new int[capacity];
+        }
+        else if (buffer.length < capacity) {
+            buffer = java.util.Arrays.copyOf(buffer, capacity);
+        }
+
+        return buffer;
+    }
+
+    public static long[] ensureCapacity(long[] buffer, int capacity)
+    {
+        if (buffer == null) {
+            buffer = new long[capacity];
+        }
+        else if (buffer.length < capacity) {
+            buffer = java.util.Arrays.copyOf(buffer, capacity);
+        }
+
+        return buffer;
+    }
+
+    static int findNull(Block block)
```

Comment:
this method and NULL_NOT_FOUND variable are used only in DictionaryBlock; move them there and inline